### PR TITLE
Added overload to UploadFile to be able to upload file with byte[] in…

### PR DESCRIPTION
**Pull request by erzki**
Added overload to UploadFile to be able to upload file with byte[] instead of a local path.

Local path to a file is not always what you have when you uploads a file. Especially in cloud environments.

### CODE REVIEW

**Comment by saarsoo** [comment]
I can not accept this pull request unless you make the code backwards compatible.

Commit d406a7f is fine regarding the overload of the UploadFile method.

File: FortnoxAPILibrary/UrlRequestBase.cs [hunk 1]
```diff
@@ -92,14 +104,14 @@ internal string GetUrl(string index = "")
...
-            if (string.IsNullOrEmpty(ConnectionCredentials.AccessToken) || string.IsNullOrEmpty(ConnectionCredentials.ClientSecret))
+            if (string.IsNullOrEmpty(_accessToken) || string.IsNullOrEmpty(_clientSecret))
             {
                 throw new Exception("Access-Token and Client-Secret must be set");
             }
 
             HttpWebRequest wr = (HttpWebRequest)HttpWebRequest.Create(requestUriString);
-            wr.Headers.Add("access-token", ConnectionCredentials.AccessToken);
-            wr.Headers.Add("client-secret", ConnectionCredentials.ClientSecret);
+            wr.Headers.Add("access-token", _accessToken);
```
Comment (saarsoo) [inline FortnoxAPILibrary/UrlRequestBase.cs]: You could set the ConnectionCredentials to be the back up value if the credentials wasn't specified during the creation of the connector. To maintain backwards compatibility an empty constructor would be required for the UrlRequestBase class and the classes that inherit from the UrlRequestBase class.

How about something like this?

``` C#
wr.Headers.Add("access-token", _accessToken ?? ConnectionCredentials.AccessToken);
```

Comment (erzki) [inline FortnoxAPILibrary/UrlRequestBase.cs]: Yes, that would work for backward compatibility. You fix?

Comment (saarsoo) [inline FortnoxAPILibrary/UrlRequestBase.cs]: You will have to change your pull request if you want this code to merged.

File: FortnoxAPILibrary/UrlRequestBase.cs [hunk 2]
```diff
@@ -92,14 +104,14 @@ internal string GetUrl(string index = "")
         internal HttpWebRequest SetupRequest(string requestUriString, string method)
         {
             Error = null;
-            if (string.IsNullOrEmpty(ConnectionCredentials.AccessToken) || string.IsNullOrEmpty(ConnectionCredentials.ClientSecret))
+            if (string.IsNullOrEmpty(_accessToken) || string.IsNullOrEmpty(_clientSecret))
```
Comment (saarsoo) [inline FortnoxAPILibrary/UrlRequestBase.cs]: This condition is currently only checking the custom parameters, using the default ConnectionCredentials will yield an exception whenever a request is setup now.

**Comment by davidulvmoen** [comment]
I agree @erzki! FileUpload is not in our current roadmap, but feel to get back if this is still valid.

**Pull request closed without merging**